tiledb/sm/stats/stats.h

```diff
@@ -146,10 +153,10 @@ class Stats {
    * A pointer to the parent instance that manages this
    * lifetime of this instance.
    */
-  Stats* parent_;
+  tdb_shared_ptr<Stats> parent_;
```

Comment:
@eric-hughes-tiledb correct me if I am wrong:
`parent_` should be a `week_ptr `to avoid cyclic dependency that will lead to Stats never getting freed

_Note_: there is currently no tdb_weak_ptr?

Reply:
If things are actually arranged in a tree, there shouldn't be an actual issue. It's a useful pattern, indeed, to have a child hold a strong pointer (lifespan-preserving) to parent so that the child has an assurance that the parent is always present. If the parent needs to hold a pointer to its children, that's where you want a weak pointer.

There is no tdb_weak_ptr. After the new allocator is merged, you can use `std::weak_ptr`, since `tdb_shared_ptr` is changing to an alias for `std::shared_ptr`. Not wanting to reimplement yet-another standard library class was my initial motivations for the allocator work.